src/components/AddTechDialog.tsx

```diff
@@ -0,0 +1,177 @@
+import React, {  useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { makeStyles, Theme, createStyles, FormControl, Grid } from '@material-ui/core';
+import { useSnackbar } from 'notistack';
+import { addTech } from '../api/techs/addTech';
+import { User } from '../models/user';
+import { TextField } from '@material-ui/core';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    form: {     
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "center",
+    },
+    addButton: {
+      margin: theme.spacing(2),
+    },
+    formControl: {
+        minWidth: 160,
+    },
+  })
+);
+
+export interface AddTechDialogProps {
+  setTechs: (value: React.SetStateAction<User[]>) => void;
+}
+
+const AddTechDialog = (props: AddTechDialogProps) => {
+  const classes = useStyles();
+  const [open, setOpen] = React.useState(false);
+
+  const [username, setUsername] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [confirmPassword, setConfirmPassword] = useState<string>('');
+
+  const [submitted, setSubmitted] = useState<boolean>(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+    setSubmitted(false);
+  }
+
+  const handleClickClose = () => {
+    setOpen(false);
+
+    setUsername('');
+    setPassword('');
+    setConfirmPassword('');
+    setSubmitted(false);
+  }
+
+  const { enqueueSnackbar } = useSnackbar();
```

Comment:
move it to the very top of the component.

Reply:
Done!